### PR TITLE
SE hard fork instructions

### PR DESCRIPTION
**Pull request by brentstone**

### CODE REVIEW

File: HardFork.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Validator instructions for hard fork and state migration
+
+The hard fork is planned to take place at block height `X`. The following instructions to accomplish this are detailed below.
+
+## Before the hard fork
+
+1. Shut down your node and restart it using `namadan ledger run-until --block-height X --suspend`. This will run your ledger and suspend it automatically at the provided block height `X`, where the hard fork is planned.
+2. Once you reach height `X`, you may have to shut down your node with `Ctrl+C`.
+3. Backup your state by zipping your `.local/share/namada/shielded-expedition.88f17d1d14` directory.
```
Comment (Fraccaman) [inline HardFork.md:9]: The base directory might not be the default one

**Comment by Fraccaman** [comment]
this looks great! we should also provide for some of those steps, an example output so that people can tell if the command was executed correctly or not.

File: HardFork.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+### Getting `migrations.json`
+
+Node operators should attempt to produce the `migrations.json` file themselves. At this moment, this requires building Namada from source somewhere. Please refer to https://github.com/anoma/namada/blob/main/README.md for instructions on how to do this.
+
+You will need to build namada from source using [#2937](https://github.com/anoma/namada/pull/2937), which is the branch `brent/test-migration-from-0.31.9`. This is also the same branch that `namadan-0.31.9-MIG` is built from.
+
+Given that you have built namada from source on this branch, follow these steps to produce `migrations.json`:
+
+1. Go to the root directory of your Namada built from source
+2. Run `namadan-0.31.9-MIG ledger query-db --db-key conversion_state --hash 05E2FD0BEBD54A05AAE349BBDE61F90893F09A72850EFD4F69060821EC5DE65F --db-column-family state > conversion_state.txt`
```
Comment (batconjurer) [inline HardFork.md:42]: The conversion_state.txt file must be in the top of the namada repo so the make_db_migration script can find it.

File: HardFork.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+2. Once the ledger is stopped after updating the DB, run the ledger as normal with the new binaries: `namadan-0.32.0 ledger run`.
+
+## Shielded syncing after the hard fork
+
+The hard fork will also make properly scanning for old MASP notes by running `namadac shielded-sync` a bit trickier. After the hard fork, the client will no longer be able to make RPC calls to cometBFT to query for MASP notes before block height `X`, when the hard fork takes place. There are several solutions to help ameliorate this issue:
+
+1. Some nodes that are still running v0.31.9 of the software are listed at `<insert_URL>`. You can query these nodes with `namadac shielded-sync` to scan for all MASP notes up until height `X`.
+2. In v0.32.0, `shielded-sync` has a new option `--from-height` that will start the MASP note scanning at the given height. When you query from a v0.32.0 node, please for at least the first time, use `namadac shielded-sync --from-height X`.
```
Comment (batconjurer) [inline HardFork.md:62]: The precise conditions to be able to remove the `--from-height` flag are as follows:
 1. The user must have keys in their shielded context
 2. They successfully scan a masp tx with this command (if there are none, they will keep having to pass `--from-height X`.)

File: HardFork.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+1. Run `namadan-0.31.10 ledger query-db --db-key conversion_state --hash 05E2FD0BEBD54A05AAE349BBDE61F90893F09A72850EFD4F69060821EC5DE65F --db-column-family state > conversion_state.txt`. Make sure the produced `conversion_state.txt` file is in your current working directory.
+2. Make a direcory called `wasm` in your current directory and then copy the new `vp_user.6065919f895d43099a567cb120ebdfa0c99c3ba4e803fe99159a14bd8f97f0ea.wasm` and `checksums.json` files from [hard_fork directory](hard_fork/) into `wasm/`. This might involve remote copying the files onto your machine and then performing:
+    ```
+    mkdir wasm
+    mv <some_path>/vp_user.6065919f895d43099a567cb120ebdfa0c99c3ba4e803fe99159a14bd8f97f0ea.wasm wasm/
+    mv <some_path>/checksums.json wasm/
+    ```
+3. Run `./make-db-migration` and confirm that `migrations.json` is produced in your current directory as a result. You may have to first run `chmod +x make-db-migration` to make the binary executable.
```
Comment (batconjurer) [inline HardFork.md:41]: run from you working directory

**Pull request merged**